evm_loader/program/src/entrypoint.rs

```diff
@@ -704,17 +731,29 @@ fn do_continue<'a>(
         }
     };
 
+    Ok(Some(call_results))
+}
+
+fn applies_and_invokes<'a>(
+    program_id: &Pubkey,
+    account_storage: &mut ProgramAccountStorage<'a>,
+    accounts: &'a [AccountInfo<'a>],
+    operator: Option<&AccountInfo<'a>>,
+    call_results: SuccessExitResults
+) -> ProgramResult {
+    let (exit_reason, used_gas, result, applies_logs_transfers) = call_results;
```

Comment:
Moved `account_storage.apply*` and invoke on logs and result outside `do_call` and `do_continue`  to separate function to force former inner instructions order.